src/components/visualizer/variables.tsx

```diff
@@ -35,7 +29,7 @@ function VarTable({ vars, title }: VarTableProps): React.ReactElement {
     });
 
     setTableValues(trs);
-  }, vars);
+  }, [vars]);
```

Comment:
The functionality of this change needs to be tested to see if the desired original functionality is still present.

Desired: on change of the array, or any elements in the array, reEvaluate the table

Reply:
React errored for the original code because vars changes length, which is disallowed. The current code works because vars is never mutated (always recreated), so reference equality works fine. If we changed the code to mutate vars at some point in the future, there would be a problem.

Also, this is somewhat unrelated, but I'm unfamiliar with React and wasn't familiar with the pattern you were using with useEffect. I assume it's for performance. I was interested in useMemo and found [this StackOverflow answer](https://stackoverflow.com/a/56029184), which explains some advantages of using useMemo over useEffect + useState. However, in this case, it seems like maybe we would even be able to just use React.memo to avoid re-rendering if the props don't change. I'm not totally sure: Like I said, I don't have much React experience.

Reply:
This is something to keep in mind